i6_models/parts/losses/nce.py

```diff
@@ -0,0 +1,114 @@
+__all__ = [
+    "NoiseContrastiveEstimationLossV1",
+]
+
+import torch
+from torch import nn
+from torch.nn import functional as F
+from typing import Optional
+
+
+class NoiseContrastiveEstimationLossV1(nn.Module):
+    __constants__ = ["num_samples", "log_norm_term", "reduction"]
+    num_samples: int
+    log_norm_term: float
+
+    def __init__(
+        self,
+        num_samples: int,
+        *,
+        model: nn.Module,
+        noise_distribution_sampler: nn.Module,
+        log_norm_term: Optional[float] = None,
+        reduction: str = "none",
+        device: Optional[str] = None,
+    ) -> None:
+        """
+        Noise contrastive estimation (NCE) loss implementation.
+        Used to estimate the softmax. Normally for very large softmax sizes, for example word-level LM.
+
+        :param num_samples: num of samples for the estimation, normally a value between 1000-4000.
+                            2000 is a good starting point.
+        :param model: model on which the NCE loss is to be applied. The model requires a member called `output`, which
```

Comment:
Why not pass in these two weights explicitly?

Reply:
hmmm I think that would be possible... I dont have a strong preference either way.. do you prefer explicitly passing of the weights?
@Gerstenberger or is there a specific reason why you pass the model and not the weights?

Reply:
Passing the weights is fine.
There is no particular reason for passing the model instead. Initially, i did pass the weights itself but changed that for some reason i cannot remember.

Reply:
Let's pass the weights then directly. This way you don't need to have the model follow some convention.